doc/configuration.rst

```diff
@@ -598,6 +598,42 @@ examples into a single mini-gallery, e.g.:
 For such a mini-gallery, specifying a custom heading message is recommended
 because the default message is vague: "Examples of one of multiple objects".
 
+Create mini-galleries using file paths
+""""""""""""""""""""""""""""""""""""""
+Sometimes you may want to explicitly create a mini-gallery for files that may be 
+in the same location but not have functions in common, for example a set of 
+tutorials. The mini-gallery directive supports passing in a string path and 
+a glob-style string to a set of files. 
+
+.. code-block:: rst
+
+    .. minigallery:: numpy.exp ../examples/plot_0_sin.py ../examples/plot_4*
+
+
+.. minigallery:: numpy.exp ../examples/plot_0_sin.py ../examples/plot_4*
+   :add-heading: Mini-gallery using paths
+   :heading-level: -
+
+The ordering of thumbnails can be set using the ``minigallery_sort_order`` 
+configuration key. This configuration key supports any callable that is a valid 
+*key* argument to ``sorted`` and is provided two parameters to sort on, 
+the input ``obj`` and the filepath ``path``:
+
+.. code-block:: rst
+    
+    sphinx_gallery_conf = {
+    ...,
+     "minigallery_sort_order": lambda x: (
+        not (x[0] == "numpy.exp"),
+        not x[1].startswith("plot_4b"),
+        x[1],
```

Comment:
Okay one last very annoying (and tricky!) thing -- it's important that whatever is in the `sphinx_gallery_conf` have a stable `__repr__` from run to run, otherwise rebuilds will happen for all files on sequential calls to `sphinx_build` even if no files have changed. Unfortunately, we can't guarantee that `lambda`'s `__repr__` will be the same between subsequent runs of Sphinx. So really this needs to be something like:
```
class MinigallerySortOrder:

    def __repr__(self):
         return "MinigallerySortOrder"

    def __call__(self, fname):
        return ...
```
then you pass `"minigallery_sort_order": MinigallerySortOrder(),` as the conf entry. See the `warning` in https://sphinx-gallery.github.io/stable/configuration.html#sorting-gallery-subsections for example